### PR TITLE
NSA-3020-check redirect url

### PR DESCRIPTION
**Pull request by jovin-joy**

### CODE REVIEW

File: src/app/signOut/signOut.js [hunk 1]
```diff
@@ -43,5 +47,14 @@ const signUserOut = (req, res) => {
   }
 };
 
+const isValidRedirectUrl = async (url) => {
+  const getAllRedirectUrls = await servicePostLogoutStore.getServicePostLogoutRedirectsUrl(config.hostingEnvironment.serviceId);
```
Comment (DrogoNevets) [inline src/app/signOut/signOut.js:51]: doesnt catch the thrown error

File: src/app/services/servicePostLogoutStore.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            return null;
+        }
+        return await Promise.all(entities.map(async entity => ({
+            serviceId: entity.getDataValue('serviceId'),
+            redirectUrl: entity.getDataValue('redirectUrl'),
+        })));
+
+    } catch (e) {
+        logger.error(`error getting service post logout url ${id} - ${e.message}  error: ${e}`);
+        throw e;
```
Comment (DrogoNevets) [inline src/app/services/servicePostLogoutStore.js:28]: would be better to do `return Promise.reject(e)`

**Comment by DrogoNevets** [comment]
there also isnt a single test for any of the new stuff...

File: src/app/signOut/signOut.js (hunk 1 above)
Comment (jovin-joy) [reply src/app/signOut/signOut.js]: Done

File: src/app/services/servicePostLogoutStore.js (hunk 2 above)
Comment (jovin-joy) [reply src/app/services/servicePostLogoutStore.js:28]: Done

File: src/app/services/servicePostLogoutStore.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return await Promise.reject(e)
```
Comment (DrogoNevets) [inline src/app/services/servicePostLogoutStore.js:29]: this is unreachable as it is after a throw - you need to remove that line

**Pull request merged**